src/Services/KpiTypeService.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Services;
+
+use App\Entity\KpiType;
+use Doctrine\ORM\EntityManagerInterface;
+
+class KpiTypeService
+{
+    private EntityManagerInterface $entityManager;
+    private KpiTypeValidatorService $validatorService;
+
+    public function __construct(EntityManagerInterface $entityManager, KpiTypeValidatorService $validatorService)
+    {
+        $this->entityManager = $entityManager;
+        $this->validatorService = $validatorService;
+    }
+
+    public function createKpiType(array $data): array
+    {
+        try {
+            $date = new \DateTime('now', new \DateTimeZone('Asia/Dhaka'));
+
+            // Validate kpiType data
+            $validationResult = $this->validatorService->validateKpiType($data);
+
+            if (!$validationResult['status']) {
+                return $validationResult;
+            }
+
+            $kpiType = new KpiType();
+            $kpiType->setTitle($data['title']);
+            $kpiType->setSlug($data['slug']);
+            $kpiType->setStatus(1);
+            $kpiType->setCreatedAt($date);
+            $kpiType->setUpdatedAt($date);
+
+            $this->entityManager->persist($kpiType);
+            $this->entityManager->flush();
+
+            return [
+                'status' => true,
+                'message' => 'Success! The data has been added',
+            ];
+        } catch (\Exception $exception) {
+            return [
+                'status' => false,
+                'message' => $exception->getMessage(),
+            ];
+        }
+    }
+
+    public function updateKpiType(int $id, array $data): array
+    {
+        try {
+            $date = new \DateTime('now', new \DateTimeZone('Asia/Dhaka'));
+
+            // Fetch the kpiType entity to update
+            $kpiType = $this->entityManager->getRepository(KpiType::class)->find($id);
+
+            if (!$kpiType) {
+                return [
+                    'status' => false,
+                    'message' => 'KpiType not found',
+                ];
+            }
+
+            // Validate the updated kpiType data
+            $validationResult = $this->validatorService->validateKpiType($data);
+
+            if (!$validationResult['status']) {
+                return $validationResult;
+            }
+            // Update the kpiType entity with the new data
+            $kpiType->setTitle($data['title']);
+            $kpiType->setSlug($data['slug']);
+            $kpiType->setUpdatedAt($date);
+
+            $this->entityManager->flush();
+
+            return [
+                'status' => true,
+                'message' => 'Success! The data has been updated',
+            ];
+        } catch (\Exception $exception) {
+            return [
+                'status' => false,
+                'message' => $exception->getMessage(),
+            ];
+        }
+    }
+}
```

Comment:
Let's split this method in different methods on basis on individual task.

Reply:
Can't understand could you please clarify in detail?